java/src/jmri/jmrit/logixng/tools/swing/WhereUsedFrame.java

```diff
@@ -0,0 +1,320 @@
+package jmri.jmrit.logixng.tools.swing;
+
+import java.awt.*;
+import java.awt.event.*;
+import java.io.File;
+import java.io.IOException;
+
+import javax.swing.*;
+
+import jmri.*;
+import jmri.jmrit.entryexit.EntryExitPairs;
+import jmri.jmrit.logix.OBlockManager;
+import jmri.jmrit.logix.WarrantManager;
+import jmri.jmrit.logixng.util.WhereUsed;
+import jmri.swing.NamedBeanComboBox;
+import jmri.util.FileUtil;
+import jmri.util.swing.JComboBoxUtil;
+import jmri.util.swing.JmriJOptionPane;
+
+/**
+ * Create a where used report based on the selected bean.  The selection combo box is
+ * based on the selected type.
+ * <P>
+ * This class is a copy of jmri.jmrit.whereused.WhereUsedFrame, but it only shows
+ * where a bean is used by LogixNG. On the other hand, it shows where in the LogixNG
+ * tree the bean is used.
+ *
+ * @author Dave Sand         Copyright (C) 2020
+ * @author Daniel Bergqvist  Copyright (C) 2023
+ */
+public class WhereUsedFrame extends jmri.util.JmriJFrame {
+    ItemType _itemType = ItemType.NONE;
+    JComboBox<ItemType> _itemTypeBox;
+
+    NamedBean _itemBean;
+    NamedBeanComboBox<?> _itemNameBox = new NamedBeanComboBox<>(
+                        InstanceManager.getDefault(SensorManager.class));
+
+    JPanel _topPanel;
+    JPanel _bottomPanel;
+    JPanel _scrolltext = new JPanel();
+    JTextArea _textArea = new JTextArea();
+    JButton _createButton;
+    JLabel itemNameLabel;
+
+    public WhereUsedFrame() {
+        super(true, true);
+        setTitle(Bundle.getMessage("WhereUsed_Title"));  // NOI18N
+        createFrame();
+//        addHelpMenu("package.jmri.jmrit.whereused.WhereUsed", true);  // NOI18N
```

Comment:
I commented out this line for now. I'm not sure how to write this documentation.